lib/spack/spack/test/conftest.py

```diff
@@ -500,6 +500,25 @@ def mock_stage(tmp_path_factory: pytest.TempPathFactory, monkeypatch, request):
         shutil.rmtree(new_stage, onerror=onerror)
 
 
+@pytest.fixture(scope="session")
+def mock_stage_for_database(tmp_path_factory: pytest.TempPathFactory, monkeypatch_session):
+    """A session-scoped analog of mock_stage, so that the mock_store
+    fixture uses its own stage vs. the global stage root for spack.
+    """
```

Comment:
Is this related to this PR, or can we extract it as a bugfix in a separate PR?

Reply:
Without this change, this PR fails. No other PR I'm aware of fails. I think the way it works is to relieve contention that slows down all PR runs (but only happens to exceed thresholds in the case of this PR).